Merge develop into feature with updated package.json settings

SophiaSun18 opened the pull request.

charkour committed:
Merge pull request #272 from senior-knights/production

charkour committed:
get app running locally

charkour committed:
bump full calendar v6

charkour committed:
remove export

charkour committed:
add engines

SophiaSun18 committed:
update to run with pnpm
remove Harmony
update About with Team 2023's information

SophiaSun18 committed:
update node version to fit the testing workflow

SophiaSun18 committed:
Update ci.yml to use pnpm commands

SophiaSun18 committed:
update ci.yml for pnpm

SophiaSun18 committed:
update cache dependency path

SophiaSun18 committed:
comment out cache for test purpose

SophiaSun18 committed:
update with new install command

SophiaSun18 committed:
Update ci.yml

NycoCC1993 committed:
Adding content to the help page

SophiaSun18 committed:
update team2023 info

SophiaSun18 committed:
Merge branch 'fix-local' of https://github.com/senior-knights/course-schedulizer into fix-local

NycoCC1993 committed:
Improving help page

SophiaSun18 committed:
Merge pull request #336 from senior-knights/fix-local

Fix local -> Develop

kvlinden committed:
testing node 18

SophiaSun18 committed:
Merge pull request #341 from senior-knights/develop

Test if develop branch can merge into Prof.Pruim's new feature branch

SophiaSun18 committed:
Merge pull request #342 from senior-knights/test/merge-feature-into-develop

Merge updated-evening-schedule into develop

NycoCC1993 committed:
Update README.md to include documentation

SophiaSun18 committed:
Merge pull request #343 from senior-knights/fhm2-patch-DocumentationIntoReadme

Update README.md to include documentation

kvlinden commented:
Is there a reason we haven't merged this branch into develop? I'm ready to do some scheduling work to help test all the changes.

SophiaSun18 commented:
> Is there a reason we haven't merged this branch into develop? I'm ready to do some scheduling work to help test all the changes.

We created another branch from Prof. Pruim's feature branch and worked with that branch last Thursday to avoid changing Prof. Pruim's code. We merged develop into the new branch then merged the new branch back to develop to ensure nothing's broken, but then it seemed like the new feature wasn't working quite right and I'm not sure if Prof. Pruim has fixed that yet. If so we can merge this PR so you can do some testing.

kvlinden commented:
Ah, I remember. I had the impression that Randy had sorted that problem out. 
Let's talk today at 3:30 and merge (probably).

NycoCC1993 committed:
Removing lint errors

NycoCC1993 committed:
Add files via upload

kcarnold committed:
Avoid caching remote schedule data.

Docs: https://developer.mozilla.org/en-US/docs/Web/API/Request/cache

NycoCC1993 committed:
Removing extraneous alteration messages

SophiaSun18 committed:
Unused capacity-related sections removed

SophiaSun18 committed:
Term -> Academic Period

SophiaSun18 committed:
Department -> Parse the first piece of CourseSection

SophiaSun18 committed:
Prefix, Number, Section, Course Title -> Parse everything from Course Section, but didn't comment out previous cf functions

SophiaSun18 committed:
Faculty Hours, Student Hours -> Semester Hours

SophiaSun18 committed:
BuildingAndRoom -> Location 1 & 2, Using the same cf function but only creating meeting sections for 1

SophiaSun18 committed:
Meeting Day, Meeting Time -> Meeting Patterns

SophiaSun18 committed:
Split part of original courseSectionCallback() into courseCallback()
to fix the error of "cannot read properties of undefined"

SophiaSun18 committed:
SemesterStartDate, SemesterEndDate -> StartDate, EndDate
original StartDate (column name: TermStart) commented out: same value as SemesterStartDate, all combined into the StartDate column of new format
Semester Length = EndDate - StartDate

NycoCC1993 committed:
Merge pull request #348 from kcarnold/bug/bust-cache

Avoid caching remote schedule data. Add only this element to develop.

SophiaSun18 committed:
Fixed `readFullCSV.spec.ts` to pass `pnpm test` using updated example schedule

SophiaSun18 committed:
Fixed minor meetingTime issue to accomodate the most recent schedule

SophiaSun18 committed:
Update meetingPatternCallback() and timeCallback() to process sections with multiple meeting times

SophiaSun18 committed:
Update output format

SophiaSun18 committed:
Update testing script

SophiaSun18 committed:
Update deploy.yml for `production` deployment

NycoCC1993 committed:
Adding changes

NycoCC1993 committed:
Fix jest-dom version needs extend-expect

NycoCC1993 committed:
Merge pull request #349 from senior-knights/feature/update-scheduling-format

Feature/update scheduling format

NycoCC1993 committed:
Merge pull request #350 from senior-knights/update/develop

Merging code from both Sophia and Faeren's branches into the Develop Branch

SophiaSun18 committed:
update package.json

NycoCC1993 committed:
Update package.json

SophiaSun18 committed:
Merge pull request #351 from senior-knights/develop_repair_small_01

Update package.json

SophiaSun18 committed:
Update about page to include 2024 final report and presentation slides

SophiaSun18 committed:
update deploy.yml with pnpm install and update

SophiaSun18 committed:
Merge pull request #353 from senior-knights/develop-update-about-2024

Develop update about 2024

kvlinden committed:
Cleaning up internal and external documentation.
- Moved external user documentation to the About and Help pages.
- Distributed internal documentation between the root README (project-general development philosophy) and the Schedulizer client README (client-specific workflows)

kvlinden committed:
Sorted out Node/pnpm versioning

kvlinden committed:
Reverting to develop configuration to test deployment

kvlinden committed:
Merge pull request #356 from senior-knights/fix/documentation

Documentation cleanup and deployment test

kvlinden committed:
finished documentation update

kvlinden committed:
Merge pull request #357 from senior-knights/fix/documentation

Finished updating internal and external documentation

kvlinden committed:
More configuration cleanup
- configured Corepack and pnpm to set the correct versions of Node and pnpm
- made final edits to the client workflow documentation
- fixed the versions for 3 npm packages so that the system compiles and runs
- fixed Sam's image
All the npm package versions will need to be revisited and, perhaps, updated.

kvlinden committed:
Merge pull request #358 from senior-knights/fix/tool-versions

Fixed versioning issues

wahrheit319 committed:
Added 2024 team members and updated AboutPage

edomamaru committed:
Merge pull request #359 from senior-knights/feature/add-2024-team-members

add 2024 team members

wahrheit319 committed:
added 2024 proposal and proposal and status report

edomamaru committed:
Merge pull request #360 from senior-knights/feature/add-2024-team-members

added 2024 proposal and status report

wahrheit319 committed:
fixed the AboutPage link

neh8 committed:
Merge pull request #361 from senior-knights/feature/add-2024-team-members

correct the AboutPage link

wahrheit319 committed:
fixed issue #190

wahrheit319 committed:
now able to read in new format

neh8 committed:
Updated @types/papaparse and @testing-library/user-event

neh8 committed:
Updated @types/papaparse (successful 2nd attempt); @types/yup; @types/lodash to 4.14.190 (highest it can go without errors)

neh8 committed:
Updated eslint-plugin-import, eslint-plugin-jsx-ally, and eslint-plugin-react

neh8 committed:
Removed @types/yup because it is deprecated and unnecessary since yup has its own type definitions.

neh8 committed:
Updated papaparse, sass, and yup

neh8 committed:
Updated all @mui packages and @testing-library packages

neh8 committed:
Updated @types/jest, @types/node, @types/object-hash, and @types/react-stickynode

neh8 committed:
Updated react-spring to 9.7.4; react-stick and react-stickynode to latest; walked back @types/react-dom because it has an error

neh8 committed:
Latest version of lockfile

neh8 committed:
Restored @testing-library/jest-dom. Despite passing all manually-run tests and compiling without errors, upgrading the version of this package seems to be causing GitHub tests to fail.

neh8 committed:
Updated eslint-config-airbnb, eslint-plugin-react-hooks, and eslint-plugin-typescript-sort-keys

neh8 committed:
Updated @typescript-eslint/eslint-plugin

neh8 committed:
Updated lint-staged

neh8 committed:
Updated gh-pages

neh8 committed:
Updated husky

neh8 committed:
Updated @types/react-dom to latest working version (16.14.62)

neh8 committed:
Removed @types/testing-library__react because it is deprecated and because testing-library__react provides its own type definitions

neh8 committed:
Updated @types/react-dom to latest working version (18.3.5)

neh8 committed:
Updated @typescript-eslint/parser to highest working version (5.61.0)

neh8 committed:
Updated eslint to highest working version (8.57.0). Note that this is still a deprecated version and will be listed as such when you run pnpm install.

wahrheit319 committed:
new button EXPORT EXCEL that does nothing yet

wahrheit319 committed:
now it exports an empty xlsx file

wahrheit319 committed:
Exports like import now, but missing some columns, fix later.

wahrheit319 committed:
tried to fix missing values

wahrheit319 committed:
More columns are appearing, but they need more fixes

wahrheit319 committed:
change back the time format

wahrheit319 committed:
Now XLSX export matches import, but need more checks

wahrheit319 committed:
updated packag.json from update/versions

wahrheit319 committed:
update pnpm-lock.yaml

wahrheit319 committed:
Merge pull request #369 from senior-knights/update/versions

Update/versions

wahrheit319 committed:
it can now export a second spreadsheet showing export time.

wahrheit319 committed:
fix maximum credit

wahrheit319 committed:
fix MeetingDays format, like how it was imported

wahrheit319 committed:
update Registrar Schedule on second sheet

wahrheit319 committed:
fix reading old CSV files and added new testing csv/xlsx in csv folder

neh8 committed:
Updated @mui packages to latest (0.0.1) version that was recently released

neh8 committed:
Merge branch 'update/versions' of github.com:senior-knights/course-schedulizer into update/versions

neh8 committed:
Test commit (updated sass)

neh8 committed:
Updated eslint-config-prettier to latest working version (7.2.0)

neh8 committed:
Updated material-ui-popup-state to latest working version (3.1.1)

neh8 committed:
Updated prettier

neh8 committed:
Updated runtypes to latest working version (6.7.0)

neh8 committed:
Updated typescript to latest working version (4.7.4)

wahrheit319 committed:
remove extra fields

wahrheit319 committed:
updated version updates from update/version, wow

wahrheit319 committed:
Merge branch 'feature/new-format' into update/versions

neh8 committed:
Merge pull request #370 from senior-knights/update/versions

Update/versions

neh8 committed:
Added data structure diagram

neh8 committed:
Added diagram of data structure

kvlinden committed:
Merge pull request #371 from senior-knights/add-diagram

Added diagram of data structure

kvlinden committed:
Merge pull request #365 from senior-knights/feature/new-format

Feature/new-format

wahrheit319 committed:
added comment and delivery mode columns, a dropdown to pick Online, In-Person, or Hybrid.

wahrheit319 committed:
changed drop down order to In-Person, Online, Hybrid.

wahrheit319 committed:
Added Meta tab with current date and time display

neh8 committed:
Updated a few minor versions that came out recently: @types/node, prettier, @typescript-eslint/eslint-plugin, @mui packages, lint-staged, eslint-plugin-react-hooks, sass

neh8 committed:
Updated react-spring

wahrheit319 committed:
update/version

neh8 committed:
It appears the Lodash package, which previously threw errors, is now able to be updated.

wahrheit319 committed:
update/versions

wahrheit319 committed:
revert version/updates for later

wahrheit319 committed:
fixed missing column values after update

neh8 committed:
Updated minor @mui and @types/react versions that came out this weekend.

wahrheit319 committed:
fixed CourseLevel after update

wahrheit319 committed:
mutiple test

wahrheit319 committed:
show file names on toggle

wahrheit319 committed:
prevent accidentally uncheck all schedules and lose all changes

wahrheit319 committed:
fixed toggle schedules disappearing when updating section

wahrheit319 committed:
added a button for TOGGLE SCHEDULES

wahrheit319 committed:
compare test

wahrheit319 committed:
fixed: prevent automatic schedule merging after updates

wahrheit319 committed:
added a test file for compare function

wahrheit319 committed:
reordered DeliveryMode and Comment column in excel export

kvlinden committed:
Merge pull request #374 from senior-knights/feature/delivery-mode-interface

Feature/delivery mode interface

wahrheit319 committed:
reordered DeliveryMode and Comment column in excel export

wahrheit319 committed:
fixed update section logic

wahrheit319 committed:
added truncates sheet names to avoid exceed excel's 31 character limit

wahrheit319 committed:
fixed export comparison columns

wahrheit319 committed:
added ability to track and display which columns changed during updates

wahrheit319 committed:
I think it works now

wahrheit319 committed:
small version update

wahrheit319 committed:
fixed AcademicYear and added more toggle button for comparison

wahrheit319 committed:
better ui for compare schedule

wahrheit319 committed:
fixed a grey line in compare schedule tab

wahrheit319 committed:
fixed remove section logic and issue of auto merging schedules after update

wahrheit319 committed:
some test files

wahrheit319 committed:
added grouped faculty sheet, probably wrong

wahrheit319 committed:
fixed show difference logic

wahrheit319 committed:
better ui for compare

wahrheit319 committed:
better UI for toggle

wahrheit319 committed:
remove extra title

wahrheit319 committed:
usable meta tab with versions, years, and notes

wahrheit319 committed:
fixed meta

wahrheit319 committed:
better ui in compare

wahrheit319 committed:
better compare ui

wahrheit319 committed:
ui fixes

wahrheit319 committed:
better ui

wahrheit319 committed:
test file

wahrheit319 committed:
added a save button for meta tab

wahrheit319 committed:
added auto save for meta

neh8 committed:
Updated Typescript

wahrheit319 committed:
automatically update preview in ScheduleComparer after column selection changes

wahrheit319 committed:
update/versions typescript and runtypes

neh8 committed:
Updated typescript.

neh8 committed:
Merge branch 'newbranch' into update/versions

neh8 committed:
Overwrite update/versions with content from the develop branch

kvlinden committed:
Merge pull request #377 from senior-knights/update/versions

Update/versions

kvlinden committed:
updating the help for the new XLSX format features

kvlinden committed:
Merge pull request #379 from senior-knights/update/help

Updating the help for the new XLSX format features

kvlinden committed:
updated schedule editing help instructions

kvlinden committed:
removed formatting field list
streamlined the help

kvlinden committed:
removed the reference to the old formatting field recommendations

kvlinden committed:
removed reference to CSV schedule files
removed (inoperative) demo button

kvlinden committed:
Merge pull request #381 from senior-knights/update/help

updated schedule editing help instructions

kvlinden committed:
Resetting CI so that warnings are not treated as errors on pnpm build

kvlinden committed:
changing to Gihub tokens

wahrheit319 committed:
fixed URL argument

wahrheit319 committed:
fixed conflicts

wahrheit319 committed:
fixed conflict

kvlinden committed:
Merge pull request #376 from senior-knights/feature/compare

Good - This will be a good demo branch for the final presentation and a good starting point for next year's team.

neh8 committed:
Added final report and presentation slides

kvlinden committed:
Merge pull request #388 from senior-knights/addMaterialsAgain

Added final report and presentation slides
